### PR TITLE
finalise `esConsumes` migration of HLT pkgs

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR is part of #31061.

It should complete the `esConsumes` migration of packages that require a signature from HLT.

It includes minor cleanup of a couple of unrelated modules.

Merely technical. No changes expected.

#### PR validation:

`scram` builds.

#### If this PR is a backport, please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
please test

Puzzling error in [a non-HLT `addOnTest`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c798ef/20488/addOnTests/fastsim1/cmsDriver.py_TTbar_13TeV_TuneCUETP8M1_cfi_--conditions_auto:run2_mc_l1stage1_--fast__-n_100_--eventcontent_AODSIM,DQM_--relval_100000,1000_-s_GEN,SIM,.log). At the moment, I can't reproduce the issue locally with this PR, and the problem does not come from the IB. I'll ask to rerun the tests, while I try to figure out what happened.

**Comment by missirol** [comment]
The [original error in the `addOnTests`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c798ef/20488/addOnTests/fastsim1/cmsDriver.py_TTbar_13TeV_TuneCUETP8M1_cfi_--conditions_auto:run2_mc_l1stage1_--fast__-n_100_--eventcontent_AODSIM,DQM_--relval_100000,1000_-s_GEN,SIM,.log) is now gone, so indeed most likely unrelated to this PR.

Don't know if this type of issue happened before. Tagging @makortel in case this is of interest to him.

**Comment by makortel** [comment]
Thanks @missirol, seems something to be reported in https://github.com/cms-sw/cmssw/issues/24051 (I'll do).

**Comment by missirol** [comment]
+hlt

**Comment by jfernan2** [comment]
+1

**Comment by rekovic** [comment]
+1

**Comment by perrotta** [comment]
+1

File: DQM/HLTEvF/plugins/TriggerBxMonitor.cc [hunk 1]
```diff
@@ -127,6 +125,7 @@ void TriggerBxMonitor::fillDescriptions(edm::ConfigurationDescriptions& descript
 
 TriggerBxMonitor::TriggerBxMonitor(edm::ParameterSet const& config)
     :  // module configuration
+      m_l1tMenuToken{esConsumes()},
```
Comment (Dr15Jones) [inline DQM/HLTEvF/plugins/TriggerBxMonitor.cc:128]: I believe bookHistogram is called on begin Run transitions, therefore this esConsumes call must also say it happens on begin Run.

File: DQM/HLTEvF/plugins/TriggerRatesMonitor.cc [hunk 2]
```diff
@@ -159,6 +157,7 @@ void TriggerRatesMonitor::fillDescriptions(edm::ConfigurationDescriptions &descr
 
 TriggerRatesMonitor::TriggerRatesMonitor(edm::ParameterSet const &config)
     :  // module configuration
+      m_l1tMenuToken{esConsumes()},
```
Comment (Dr15Jones) [inline DQM/HLTEvF/plugins/TriggerRatesMonitor.cc:160]: This also needs to be marked as begin Run

**Comment by missirol** [comment]
Thanks @Dr15Jones (and apologies for missing this). Will open a PR to fix this.

**Pull request merged**